starter-code/MyLinkedList.java

```diff
@@ -1,62 +1,101 @@
-package ListLab;
-
 import java.util.Iterator;
 
-public class MyLinkedList<T> implements Iterable<T>{
-
-	private Node<T> head;
-	private int size = 0;
-
-	public int getSize() {
-		return size;
-	}
-
-	public boolean remove(int index) {
-		//to-do
-	}
-
-	public T get(int index) {
-		//to-do
-	}
-
-	public void add(T obj) {
-		//to-do
-	}
-
-	//The methods below are bonus
-
-	public void add(int index, T obj){
-		//to-do
-	}
-
-	@Override
-	public Iterator<T> iterator() {
-		// TODO Auto-generated method stub
-		return new MyIterator<T>(head);
-	}
-
-	private class MyIterator<T> implements Iterator<T> {
-
-		private Node<T> node;
-
-		public MyIterator(Node<T> node) {
-			//to-do
-		}
-		@Override
-		public boolean hasNext() {
-			//to-do
-		}
-
-		@Override
-		public T next() {
-			//to-do
-		}
-
-		@Override
-		public void remove() {
-			// to-do
-
-		}
-
-	}
+public class MyLinkedList<T> implements Iterable<T> {
+
+    private Node<T> head;
+    private int size = 0;
+
+    public int getSize() {
+        return size;
+    }
+
+    public boolean remove(int index) {
+        if (size == 0) {
+            return false;
+        }
+        if (index > size - 1) {
+            return false;
+        }
+        if (index == 0) {
+            head = head.getNext();
+            return true;
```

Comment:
Good call checking for `index == 0`! This is important and easy to overlook, so I'm glad you considered it :)